server/src/handlers/users.rs

```diff
@@ -7,61 +7,262 @@ use axum::{
 use futures::TryStreamExt;
 use mongodb::{
     bson::oid::ObjectId,
-    results::{DeleteResult, InsertOneResult, UpdateResult},
+    results::{DeleteResult, UpdateResult},
 };
+use serde::{Deserialize, Serialize};
 
+use crate::auth::middleware::{AdminAuth, AuthorizationError, SelfOrAdminAuth};
+use crate::models::Token;
 use crate::{
+    auth,
     db_connection::DBClient,
     error::CheeseError,
     models::{User, UserList},
 };
 
 pub async fn fetch_users(
     Extension(db_client): Extension<DBClient>,
+    _auth: AdminAuth,
 ) -> Result<Json<UserList>, CheeseError> {
     let cursor = db_client.user_repo.get_all_users().await?;
-    Ok(Json(UserList {
-        users: cursor.try_collect().await?,
-    }))
+
+    let mut users: Vec<User> = cursor.try_collect().await?;
+    for u in &mut users {
+        u.hashed_password = String::new();
```

Comment:
I think the tokens are now still returned?
Maybe a generic `SanitizedUser` struct, with a `User::sanitize(self) -> SanitizedUser` might be nice.

Reply:
fixed this and added a test for it in the new branch. Will also add sanitized users in a future PR, but that won't be a breaking change.